pageserver/src/lib.rs

```diff
@@ -168,18 +169,36 @@ impl PageServerConf {
 
 /// External relish storage configuration, enough for creating a client for that storage.
 #[derive(Debug, Clone)]
-pub enum RelishStorageConfig {
-    /// Root folder to place all stored relish data into.
+pub struct RelishStorageConfig {
+    /// Limits the number of concurrent sync operations between pageserver and relish storage.
+    pub max_concurrent_sync: usize,
+    /// The storage connection configuration.
+    pub storage: RelishStorageConnection,
+}
+
+/// A configuration to use when connecting a relish storage.
+#[derive(Debug, Clone)]
+pub enum RelishStorageConnection {
```

Comment:
Maybe `RelishStorageKind` ? `RelishStorageConnection` sounds like an active connection, like `TcpConnection`

Reply:
Hah, that's exactly what you called it in the other PR! I made my suggestion before seeing that :-).

Reply:
Funny, that's exactly what I have in the draft PR, but tried to move away from it since `kind` in rust stdlib and projects I've seen is usually a plain enum with no settings inside.

But I like kind more still, after thinking, will adjust.